core/domain/suggestion_services.py

```diff
@@ -720,3 +732,101 @@ def get_voiceover_application(voiceover_application_id):
+        suggestion_id = suggestion.suggestion_id
+        exp_id = suggestion.target_id
+        raw_latex_to_image_data_dict = (
+            suggestions_raw_latex_to_image_data_dict[suggestion_id])
+        add_svg_filenames_for_latex_strings_in_html_string = (
+            functools.partial(
+                html_validation_service.
+                add_svg_filenames_for_latex_strings_in_html_string,
+                raw_latex_to_image_data_dict))
+        suggestion.convert_html_in_suggestion_change(
+            add_svg_filenames_for_latex_strings_in_html_string)
+        updated_html_string = ''.join(suggestion.get_all_html_content_strings())
+        filenames_mapping = (
+            html_validation_service.
+            extract_svg_filename_latex_mapping_in_math_rte_components(
+                updated_html_string))
+
+        for filename, raw_latex in filenames_mapping:
+            # Some new filenames may already have the images saved from the math
+            # rich-text editor, for these files we don't need to save the image
+            # again.
+            if raw_latex in raw_latex_to_image_data_dict.keys():
+                image_file = raw_latex_to_image_data_dict[raw_latex].raw_image
+                image_validation_error_message_suffix = (
+                    'SVG image provided for latex %s failed validation' % (
+                        raw_latex))
+                try:
+                    file_format = (
+                        image_validation_services.validate_image_and_filename(
+                            image_file, filename))
+                except utils.ValidationError as e:
+                    e = '%s %s' % (e, image_validation_error_message_suffix)
+                    raise Exception(e)
+                image_is_compressible = (
+                    file_format in feconf.COMPRESSIBLE_IMAGE_FORMATS)
+                fs_services.save_original_and_compressed_versions_of_image(
+                    filename, feconf.ENTITY_TYPE_EXPLORATION, exp_id,
+                    image_file, 'image', image_is_compressible)
+
+        suggestions_to_update.append(suggestion)
+    _update_suggestions(suggestions_to_update, update_last_updated_time=False)
+
+
+def get_suggestions_with_latex_strings_having_no_svgs():
+    """Returns a batch of LaTeX strings from suggestion which have LaTeX
+    strings without SVGs.
+
+    TODO(#10045): Remove this function once all the math-rich text components in
+    suggestions have a valid math SVG stored in the datastore.
+
+    Returns:
+        dict(str, list(str)). The dict having each key as an suggestion iD and
+        value as a list of LaTeX string. Each list has all the LaTeX
+        strings in that particular suggestion.
+    """
+
+    exploration_suggestion_models = (
+        suggestion_models.GeneralSuggestionModel.get_all().filter(
+            suggestion_models.GeneralSuggestionModel.
+            target_type == 'exploration'))
```

Comment:
Use the constant here* Also should we use suggestion_type for filter?

Reply:
done